### DISCUSSION TITLE
Overlapping SEAS5 and IceNet forecast init dates

### DISCUSSION THREAD

**Discussion by EllieBowler** [icenet-ai/icenet#288] [Q&A] [answered]
Hi IceNetters! Thought i'd try using the new discussions page... but sorry if isn't the best place! I just had a question about the `compute_metrics_leadtime_avg` function. I am using it to compare average IceNet and SEAS5 forecasts over a range of initialisation dates. I have a forecast file with 36 init dates and all corresponding SEAS5 dates in the data store. But i noticed the code misses the first one out (strictly > start date but <= end date) [here](https://github.com/icenet-ai/icenet/blob/main/icenet/plotting/forecast.py#L560), so i only get 35 comparisons. I was wondering if there is a reason for this, or if the code should be changed to => for the start as well? Ta!

**Comment by bnubald** [comment]
Hi @EllieBowler 

Thanks for posting! I think this is the perfect place for it.

I've just checked some output netCDF files, and based on that, it should include the first date as you've suggested (i.e., for a prediction initialised on 2024-06-30, the first forecast day is 2024-07-01, and so, `dates >= fc_start_date` instead of `dates > fc_start_date`). It might be a bug, but would just like to confirm the versions used for (1) training, (2) prediction and (3) outputting this plot before moving forwards.

Related topic I find of use, and should definitely be documented by me...: https://github.com/icenet-ai/icenet/issues/256#issuecomment-2094923766

@EllieBowler, also, for posterity, could I ask for the following please:
- Version of IceNet.
- Command you are running.
- Location of run.
- Range of dates of SEAS data downloaded.
- Paste the output (and for ease, length) of the `forecast_date` variable in the `forecast_file` netCDF file being passed to the `compute_metrics_leadtime_avg` function.

Note: This refers to following code:

https://github.com/icenet-ai/icenet/blob/f1285368eb731ed473c6b574b1d48ae574a8c01e/icenet/plotting/forecast.py#L560

**Reply by EllieBowler** [reply]
Thanks Bryn! Yes sure:

IceNet version: 0.2.8
Command: `icenet_plot_leadtime_avg north "paper-predict/ellie_paper_north.nc" -o "plot/test.png" -m mae -dp "metrics/test.csv" -ao all -s -sm 1 -e -b -r 114,242,131,260`
Location: BAS HPC
SEAS downloaded: All currently listed in this folder: `/data/hpcdata/users/jambyr/icenet/other.data/data/mars.seas/north/siconca`
Info on forecast_date from forecast_file:
![image](https://github.com/user-attachments/assets/d394503f-38ff-4d16-8570-6e58bf2bb79d)

**Reply by bnubald** [reply, marked as answer]
Thanks @EllieBowler 

Nice find, I've just gone through this in a bit more detail, and can confirm your suspicion on it being a bug and your suggested solution.

i.e.:

`dates >= fc_start_date` instead of `dates > fc_start_date`

**Reply by bnubald** [reply]
@EllieBowler I've created corresponding issue #289
